src/main/java/com/airline/service/api/controllers/PassengerRestController.java

```diff
@@ -1,25 +1,81 @@
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
+import com.airline.service.api.entities.Flight;
+import com.airline.service.api.entities.Passenger;
+import com.airline.service.api.repos.PassengerRepository;
+import com.airline.service.api.services.FlightService;
 import com.airline.service.api.services.PassengerInfo;
 import com.airline.service.api.services.PassengerService;
 
 // Controller returns flights & passengers table as JSON
 @RestController
 public class PassengerRestController {
 	
+	@Autowired
+	private PassengerRepository passengerRepository;
 	@Autowired
 	private PassengerService passengerService;
+	@Autowired
+	private FlightService flightService;
 	
 	@GetMapping("/api/flights/passenger/{id}")
 	public PassengerInfo getPassengerInfo(@PathVariable("id") int idInfo) {
 		PassengerInfo passengerInfo = passengerService.getPassengerInfo(idInfo);
-		System.out.println(passengerInfo);
 		return passengerInfo;
 	}
+	
+	// Make reservation with external site
+	@PostMapping("/api/flights/passenger/")
+	// Requests JSON with - First name, last name, flight #
+	ResponseEntity<String> newPassenger(@RequestBody Passenger newPassenger) {
+		 HttpHeaders responseHeaders = new HttpHeaders();
+		// Attempt to grab existing flight from database, throws exception if not found
+		try {
+			 Flight flight = flightService.flightToBookById(newPassenger.getFlightNum());
+			 int seatNumber = ThreadLocalRandom.current().nextInt(1, flight.getSeatsAvailable() + 1);
+			 newPassenger.setSeatNum(String.valueOf(seatNumber));
+			 newPassenger.setBookingOrigin(1);
+			 passengerRepository.save(newPassenger);
+				
+			return new ResponseEntity<>("New passenger was successfully added.", responseHeaders, HttpStatus.OK);
+			
+		} catch(EntityNotFoundException e) {
+			System.out.println(e);
+		}
+		return new ResponseEntity<>("Failed to add passenger booking!", responseHeaders, HttpStatus.BAD_REQUEST);
+	}
+	
+	// This API endpoint can only be used to remove bookings made from external services.
+	// Attempting to remove a booking made internally will fail.
+	@DeleteMapping("/api/flights/passenger/{id}")
+	public String deletePassenger(@PathVariable("id") int idInfo) {
+		Passenger passenger = passengerRepository.findByPassengerID(idInfo);
+		
+		if(passenger.getBookingOrigin() == 1) {
+			passengerRepository.delete(passenger);
+			
+			return "Passenger deleted";			
+		} else {
+			System.out.println("Error: You may not delete a booking that did not originate from your service.");
+			throw new AccessDeniedException("Access violation occurred. Entity deletion failed.");
+		}
+		
+	}
 
```

Comment:
This looks good. Just a quick question: our booking system is more accessible for those who login and credentials are checked against our database. Will they have to login on to our service from the the package service deal to book the flight?

Reply:
They don't have to log in right now, it's an unresolved security issue. I believe it would be possible to gate the access to these APIs behind a user with a specific group permission. They can just supply their username and password in the POST body, but we probably also would need to generate API tokens for extra security. All this is probably a stretch in the time we have.